source/base/parameter_handler.cc

```diff
@@ -763,6 +763,21 @@ ParameterHandler::leave_subsection()
 
 
 
+bool
+ParameterHandler::subsection_path_exists(
+  const std::vector<std::string> &sub_path) const
+{
+  std::vector<std::string> sub_path_base(sub_path);
+
+  std::string subsection = sub_path_base.back();
+  sub_path_base.pop_back();
+
+  return static_cast<bool>(entries->get_optional<std::string>(
+    get_current_full_path(sub_path_base, subsection)));
```

Comment:
I *think* that what you are doing here is to find out whether there is an entry called `subsection` in the path `sub_path_base`, but some commentary would help. Is it guaranteed that `subsection` actually corresponds to another section, rather than just a regular entry or an alias?